spec/index.md

```diff
@@ -439,6 +439,16 @@ The state tree contains information about the topology of the Internet Computer.
         principal = bytes .size (0..29)
         tagged<t> = #6.55799(t) ; the CBOR tag
 
+-   `/subnet/<subnet_id>/metrics` (blob)
+
+     A collection of subnet-wide metrics related to this subnet's current resource usage and/or performance. The metrics are a CBOR map with the following fields:
+     
+     - `num_canisters` (`nat`): The number of canisters on this subnet.
+     - `total_canister_state` (`nat`): The total size of the state in bytes taken by canisters on this subnet since this subnet was created.
+     - `total_consumed_cycles` (`nat`): The total number of cycles consumed by canisters on this subnet since this subnet was created.
+     - `num_update_transactions` (`nat`): The total number of transactions processed on this subnet since this subnet was created.
```

Comment:
I would consider whether it makes sense to follow [Prometheus naming best practices](https://prometheus.io/docs/practices/naming/#metric-names) (or similar) when naming these metrics. Among other things, it's generally a good idea to include the measurement unit in the name (e.g. `total_canister_state_bytes` rather than a confusing `total_canister_state`). And it is good to identify counters, e,g, by giving them a `total` suffix (e.g. `total_consumed_cycles_total` or ``update_transactions_total`).

Reply:
I saw this late and the change has already been merged on the replica side. Given that these are not prometheus metrics I don't think we're bound to follow the best practices it establishes. We explain what the metrics represent so that should be ok. If spec owners feel strongly, I'm happy to change (although the hassle of changing some names given we're talking about certified state is likely not worth it imo).

Reply:
I think Alin's suggestion to make the metric names more self-explanatory makes sense so I'd be up for it if it's still possible.

Reply:
Are the following changes ok with everyone?

`total_canister_state` -> `total_canister_state_bytes`
`num_update_transactions` -> `update_transactions_total`

The suggestion of `total_consumed_cycles_total` seems confusing, at least to me (there's two instances of "total" in the name).

Reply:
I don't mind if we don't follow Prometheus naming practices specifically. The `total` can go at the end or at the start (or be replaced with something else, e.g. `all_time`). I would very much prefer it for the naming to be consistent, though. I.e. don't use `total` in both gauges and counters; and don't put it at the beginning of one metric name and the end of another.

One possible naming scheme (although, again, I'm not insisting it be followed to the letter) could be:
```
canisters
canister_state_bytes
consumed_cycles_total
update_transactions_total
```

Reply:
One technical aspect of this (which David pointed out) is that since the metrics are encoded as a blob; and the canonical state uses packed CBOR encoding; the names can be changed at any time, without breaking compatibility.

E.g. `{"num_canisters": 5, "total_canister_state_bytes": 100}` would be encoded as `{0: 5, 1: 100}`, which is the same as what `{"canisters": 5, "canister_state_bytes": 100}` would encode to (as long as we preserved field ordering).

Reply:
Consumed cycles is a gauge and update transactions is a counter, so your suggestion is inconsistent with itself if we use `total` in both. If we are ok with that, then I am ok with everything else except `canisters`, I'd prefer to keep `num_canisters` as it's more descriptive imo.

Reply:
Consumed cycles is a gauge only because it's not monotonically increasing (we subtract what are essentially reserved cycles and then return part of them later on). But leaving aside that awkwardness, it is essentially a counter, in that the absolute value is not particularly meaningful; what is meaningful about it by how much it changes over a day or an hour.

So it's all up in the air whether we want to call it a counter or a gauge. I would go with counter, since we are at very low risk of someone treating it strictly as a Prometheus counter and e.g. applying Prometheus' definition of `rate()` to it. People are just going to take the value now minus the value one hour ago. They may be surprised to see a negative difference, but that's about it.

Reply:
Alright, I buy this argument. I'll update to:

```
num_canisters
canister_state_bytes
consumed_cycles_total
update_transactions_total
```

Reply:
Updated.